gmt/utils.py

```diff
@@ -180,3 +183,64 @@ def is_nonstr_iter(value):
     except TypeError:
         is_iterable = False
     return bool(not isinstance(value, str) and is_iterable)
+
+
+class GMTTempFile():
+    """
+    Context manager for temporaray file.
+
```

Comment:
Would be good to mention that this is not a file-like object. So you can't do `for line in GMTTempFile()`, for example. Or pass it to things that need file objects.